elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchSchemaFactory.java

```diff
@@ -56,15 +61,38 @@ public ElasticsearchSchemaFactory() {
 
     try {
       final String coordinatesString = (String) map.get("coordinates");
-      Preconditions.checkState(coordinatesString != null,
-          "'coordinates' is missing in configuration");
 
-      final Map<String, Integer> coordinates = mapper.readValue(coordinatesString,
-          new TypeReference<Map<String, Integer>>() { });
+      final String hostsString = (String) map.get("hosts");
 
-      // create client
-      final RestClient client = connect(coordinates);
+      List<HttpHost> hosts;
+
+      if (coordinatesString != null) {
```

Comment:
Prefer not to use too many temp variables. I think it can be simplified to: 
```java
if (map.containsKey("hosts")) {
 hosts =   mapper.convertValue(map.get("hosts", new TypeReference<List<String>>() { })).stream() ... ;
} else if (map.containsKey("coordinates")) {
 hosts =   mapper.convertValue(map.get("coordinates", new TypeReference<Map<String, Integer>>() { })).stream() ... ;

}

Reply:
Reduced two variables.